pkg/kubelet/container/runtime.go

```diff
@@ -84,6 +84,8 @@ type Runtime interface {
 	ContainerCommandRunner
 	// ContainerAttach encapsulates the attaching to containers for testability
 	ContainerAttacher
+	// Get Container NetNamespacePath
+	GetNetNs(containerId string) (string, error)
```

Comment:
To be clearer: This comment meant that CNI lives entirely within the docker-specific adapter in our Kubelet.  Not in Docker.

The point being that if you run rkt, you want none of what this PR is pushing because rkt has its own network management and it understands pods already, and besides that it doesn't have the same hook that we use (the POD container).

If you run docker you have a choice (eventually): let kubernetes do network setup (via CNI) or let docker do network setup (eventually via libnetwork).  If, at some point in the future, libnetwork is actually viable maybe we won't need CNI here.  Or maybe we'll have a libnetwork->CNI driver.  Who knows.